src/staking/block_validator.cpp

```diff
@@ -248,6 +300,15 @@ class BlockValidatorImpl : public AbstractBlockValidator {
     if (validation_info.GetHeight() != static_cast<blockchain::Height>(prev_block.nHeight) + 1) {
       return BlockValidationResult(Error::MISMATCHING_HEIGHT);
     }
+    std::int64_t lock_time_cutoff = prev_block.GetMedianTimePast();
```

Comment:
Do we already have tests covering this part?

Reply:
The block validator is tested through the LegacyValidationInterface which has a New (goal) and a Legacy (currently in use) implementation. The exact same tests are run on both the new and the old interface, see `validation_tests`. This also tests the lock time cutoff.

You wrote these tests. I extended them to also cover the new `LegacyValidationInterface` which completely delegates to the block validator.